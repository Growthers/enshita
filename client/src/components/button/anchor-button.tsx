import type { FC } from "react";
import { InlineIcon } from "@iconify/react";
import styles from "~/styles/components/button/anchor-button.module.scss";
import type {
  ButtonIconProperties,
  AnchorButtonProperties,
} from "./type/model";

const ButtonIcon: FC<ButtonIconProperties> = ({ icon, css, ...rest }) => (
  <div className={`${styles["enshita-button-icon-box"]} ${css}`}>
    <InlineIcon
      icon={icon}
      className={styles["enshita-button-icon"]}
      {...rest}
    />
  </div>
);

const AnchorButton: FC<AnchorButtonProperties> = ({
  children,
  icon,
  variant,
  iconStyles,
  boxStyles,
  textStyles,
  isSamePage,
  href,
  ...rest
}) => (
  <a
    className={`${styles["enshita-button-box"]} ${boxStyles}`}
    href={href}
    target={isSamePage ? "" : "_blank"}
    rel="noopener noreferrer"
    role="button"
    {...rest}
  >
    {variant === "icon" && <ButtonIcon icon={icon} css={iconStyles} />}
    {children && (
      <p className={`${styles["enshita-button-text"]} ${textStyles}`}>
        {children}
      </p>
    )}
  </a>
);

export { AnchorButton };
