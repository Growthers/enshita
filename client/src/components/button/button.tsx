import type { FC } from "react";
import { InlineIcon } from "@iconify/react";
import styles from "~/styles/button.module.scss";
import type { ButtonIconProperties, ButtonProperties } from "./type/model";

const ButtonIcon: FC<ButtonIconProperties> = ({ icon, css, ...rest }) => (
  <div className={`${styles["enshita-button-icon-box"]} ${css}`}>
    <InlineIcon
      icon={icon}
      className={styles["enshita-button-icon"]}
      {...rest}
    />
  </div>
);

/* eslint-disable react/button-has-type */
const Button: FC<ButtonProperties> = ({
  children,
  icon,
  variant,
  disabled,
  iconStyles,
  boxStyles,
  textStyles,
  type,
  ...rest
}) => (
  <button
    className={`${styles["enshita-button-box"]} ${boxStyles}`}
    disabled={disabled}
    tabIndex={0}
    type={type}
    {...rest}
  >
    {variant === "icon" && <ButtonIcon icon={icon} css={iconStyles} />}
    {children && (
      <p className={`${styles["enshita-button-text"]} ${textStyles}`}>
        {children}
      </p>
    )}
  </button>
);
/* eslint-enable react/button-has-type */

export { Button };
