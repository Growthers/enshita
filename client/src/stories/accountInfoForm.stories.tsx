// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
