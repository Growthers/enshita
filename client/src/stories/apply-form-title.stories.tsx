import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
