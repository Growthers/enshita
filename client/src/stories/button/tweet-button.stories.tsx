// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import TweetButton from "~/components/button/tweet-button";

type T = typeof TweetButton;
type Story = ComponentStoryObj<T>;

export default {
  component: TweetButton,
  args: {
    text: "Tweet Content",
    url: "",
    hashtags: [],
  },
  argTypes: {
    text: {
      description: "Tweetテキスト",
      controle: {
        type: "text",
      },
    },
    url: {
      description: "Tweetに含めるURL",
      controle: {
        type: "text",
      },
    },
    hashtags: {
      description: "Tweetに含めるハッシュタグ",
      controle: {
        type: "object",
      },
    },
  },
} as ComponentMeta<T>;

export const Default: Story = {};

export const URL: Story = {
  args: {
    url: "https://growthers.dev",
  },
};

export const Hashtags: Story = {
  args: {
    hashtags: ["共同開発鯖", "#共同開発鯖LT"],
  },
};

export const Full: Story = {
  args: {
    url: "https://growthers.dev",
    hashtags: ["共同開発鯖", "#共同開発鯖LT"],
  },
};
