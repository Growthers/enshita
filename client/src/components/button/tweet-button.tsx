import React from "react";
import { AnchorButton } from "~/components/button/anchor-button";
import styles from "~/styles/components/button/tweet-button.module.scss";

type Props = {
  text: string;
  url: string | undefined;
  hashtags: string[] | undefined;
};

const TweetButton: React.FC<Props> = ({ text, url = "", hashtags = [] }) => {
  // URL, hashtagsがないときはパラメータの値がなにもないが、問題なく指定した項目のみでTweet可能
  const tweetLink = `https://twitter.com/intent/tweet?text=${text.replace(
    / /g,
    "%20",
  )}&url=${url}&hashtags=${
    hashtags ? hashtags.map(v => v.replace("#", "")).join(",") : ""
  }`;

  return (
    <AnchorButton
      variant="icon"
      href={tweetLink}
      icon="akar-icons:twitter-fill"
      iconStyles={styles["enshita-event-tweet-icon"]}
      boxStyles={styles["enshita-event-tweet-box"]}
      textStyles={styles["enshita-event-tweet-text"]}
    >
      Tweet
    </AnchorButton>
  );
};

export default TweetButton;
