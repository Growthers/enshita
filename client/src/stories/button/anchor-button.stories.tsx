import type { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import { AnchorButton } from "~/components/button/anchor-button";

type T = typeof AnchorButton;
type Story = ComponentStoryObj<T>;

const data = {
  sentence: "This is test",
  link: "#",
};

export default {
  args: { children: data.sentence, variant: "normal", href: data.link },
  component: AnchorButton,
} as ComponentMeta<T>;

export const Default: Story = {};

export const Icon: Story = {
  args: {
    variant: "icon",
    icon: "akar-icons:twitter-fill",
  },
};
