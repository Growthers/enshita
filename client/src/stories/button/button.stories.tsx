import type { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import { Button } from "~/components/button/button";

type T = typeof Button;
type Story = ComponentStoryObj<T>;

const data = {
  sentence: "This is test",
};

export default {
  args: {
    children: data.sentence,
    type: "button",
    disabled: false,
    variant: "normal",
  },
  component: Button,
} as ComponentMeta<T>;

export const Default: Story = {};

export const Disabled: Story = {
  args: {
    disabled: true,
  },
};

export const Icon: Story = {
  args: {
    variant: "icon",
    icon: "akar-icons:twitter-fill",
  },
};
