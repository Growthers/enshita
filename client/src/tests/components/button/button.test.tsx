import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/button/button.stories";

const { Default, Disabled, Icon } = composeStories(stories);

describe("(components) button", () => {
  const options = { name: "This is test" };
  test("[role=button]である", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("button", options)).toBeInTheDocument();
  });
  test("[disabled=true]である", () => {
    const { getByRole } = render(<Disabled />);
    expect(getByRole("button", options)).toBeDisabled();
  });
  test("(normal) take snapshot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
  test("(disable) take snapshot", () => {
    const { container } = render(<Disabled />);
    expect(container).toMatchSnapshot();
  });
  test("(icon) take snapshot", () => {
    const { container } = render(<Icon />);
    expect(container).toMatchSnapshot();
  });
});
