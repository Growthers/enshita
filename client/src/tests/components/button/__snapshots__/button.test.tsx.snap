// Vitest Snapshot v1

exports[`(components) button > (disable) test Snap Shot 1`] = `
<div>
  <button
    class="_enshita-button-box_1mmel_17 undefined"
    disabled=""
    tabindex="0"
    type="button"
  >
    <p
      class="_enshita-button-text_1mmel_36 undefined"
    >
      This is test
    </p>
  </button>
</div>
`;

exports[`(components) button > (icon) test Snap Shot 1`] = `
<div>
  <button
    class="_enshita-button-box_1mmel_17 undefined"
    tabindex="0"
    type="button"
  >
    <div
      class="_enshita-button-icon-box_1mmel_1 undefined"
    >
      <span />
    </div>
    <p
      class="_enshita-button-text_1mmel_36 undefined"
    >
      This is test
    </p>
  </button>
</div>
`;

exports[`(components) button > (normal) test Snap Shot 1`] = `
<div>
  <button
    class="_enshita-button-box_1mmel_17 undefined"
    tabindex="0"
    type="button"
  >
    <p
      class="_enshita-button-text_1mmel_36 undefined"
    >
      This is test
    </p>
  </button>
</div>
`;
